tests/integration/release/test_release.py

```diff
@@ -115,21 +145,152 @@ def test_release_row_noising(
+                    # don't noise ssa_event_type because it's used as an identifier column
+                    # along with simulant id
+                    # TODO: Noise ssa_event_type when record IDs are implemented (MIC-4039)
+                    if column == COLUMNS.ssa_event_type.name:
+                        pass
+                    else:
+                        for dataset in datasets:
+                            # noise datasets in place
+                            noise_type(dataset, config, column)
+                            with check:
+                                assert dataset.missingness.equals(
+                                    dataset.is_missing(dataset.data)
+                                )
+                        run_column_noising_test(
+                            prenoised_dataframes,
+                            datasets,
+                            config,
+                            full_dataset_name,
+                            noise_type.name,
+                            column,
+                            fuzzy_checker,
+                        )
 
-def test_column_noising(
-    unnoised_dataset: Dataset,
-    noised_data: pd.DataFrame,
-    config: dict[str, Any],
+
+def run_column_noising_test(
+    prenoised_dataframes: list[pd.DataFrame],
+    noised_datasets: list[Dataset],
+    config: NoiseConfiguration,
     dataset_name: str,
+    noise_type: str,
+    column: str,
     fuzzy_checker: FuzzyChecker,
 ) -> None:
-    """Tests that columns are noised as expected"""
-    check_noised, check_original, shared_idx = _get_common_datasets(
-        unnoised_dataset, noised_data
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset_name)
+
+    numerator = 0
+    denominator = 0
+    expected_noise_level = 0.0
+
+    # Validate column noise level
+    expected_config_noise = config.get_cell_probability(dataset_name, noise_type, column)
+    includes_token_noising = config.has_parameter(
+        dataset_name, noise_type, Keys.TOKEN_PROBABILITY, column
+    ) or config.has_parameter(
+        dataset_name, noise_type, Keys.ZIPCODE_DIGIT_PROBABILITIES, column
     )
 
-    run_column_noising_tests(
-        dataset_name, config, fuzzy_checker, check_noised, check_original, shared_idx
+    for prenoised_dataframe, noised_dataset in zip(prenoised_dataframes, noised_datasets):
+        shared_noised, shared_prenoised, shared_idx = get_common_datasets(
+            dataset_schema, prenoised_dataframe, noised_dataset.data
+        )
+
+        # Check that originally missing data remained missing
+        originally_missing_idx = shared_prenoised.index[
+            shared_prenoised.reset_index()[column].isna()
+        ]
+        with check:
+            assert shared_noised.loc[originally_missing_idx, column].isna().all()
+
+        # Check for noising where applicable
+        to_compare_idx = shared_idx.difference(originally_missing_idx)
+        if len(to_compare_idx) == 0:
```

Comment:
So once to_compare_idx is empty, we never noise but it may seem like we are noising? i think in that case we should at least log a warning or something. Im not sure how that works for users but it would be pretty confusing if I wanted to noise a dataframe and got an unnoised dataframe back because nothing was noised.